core/src/main/java/org/javarosa/xpath/expr/XPathBinaryOpExpr.java

```diff
@@ -31,13 +31,10 @@ public String toString(String op) {
     }
 
     @Override
-    public boolean equals(Object o) {
-        if (o instanceof XPathBinaryOpExpr) {
-            XPathBinaryOpExpr x = (XPathBinaryOpExpr)o;
-            return op == x.op && a.equals(x.a) && b.equals(x.b);
-        } else {
-            return false;
-        }
+    public abstract boolean equals(Object o);
```

Comment:
I don't think this is necessary, and I think it will break J2ME (or maybe it's the opposite.... It's one of the two, we will want to test a form which triggers an equality check on one of the implementing classes with the 2.27 build after it is pulled)